homeassistant/components/sensor/filter.py

```diff
@@ -277,6 +288,46 @@ def _filter_state(self, new_state):
         return filtered
 
 
+@FILTERS.register(FILTER_NAME_TIME_SMA)
+class TimeSMAFilter(Filter):
+    """Simple Moving Average (SMA) Filter.
+
+    The window_size is determined by time, and SMA is time weighted.
+    """
+
+    def __init__(self, window_size, precision, entity):
+        """Initialize Filter."""
+        super().__init__(FILTER_NAME_TIME_SMA, 0, precision, entity)
+        self._time_window = int(window_size.total_seconds())
+        self.last_leak = None
+        self.queue = deque()
+
+    def _leak(self, now):
+        """Remove timeouted elements."""
+        while self.queue:
+            timestamp, _ = self.queue[0]
+            if timestamp + self._time_window <= now:
+                self.last_leak = self.queue.popleft()
```

Comment:
Is there any reason this is assigned to an instance variable instead of a local variable and returned at the end of the while loop ?

Reply:
Please look at the previous comment https://github.com/home-assistant/home-assistant/pull/13104#discussion_r174878478

Keeping this value assures me I always have some value to work with in the _filter_state()

The filter always lags 1 state behind, so using the new state in these cases (when there are no previous states in the windows) would make the behaviour erratic.